### PR TITLE
Improve new event form

### PR DESCRIPTION
**Pull request by josh-hdz**
Added needed data processing for the back-end to accept it (change date to epoch value and price decimal formatting).

### CODE REVIEW

**Comment by josh-hdz** [comment]
I populated Datastore with 10ish events

**Review by carinegordillo — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/CreateEventServlet.java [hunk 1]
```diff
@@ -41,12 +41,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
                 .set("date", date)
                 .set("details", details)
```
Comment (valvarez1218) [inline src/main/java/com/google/sps/servlets/CreateEventServlet.java:42]: Shouldn't this also be "description" as you change it above?

File: src/main/webapp/createEvent.html [hunk 2]
```diff
@@ -48,16 +48,16 @@ <h1 class=center>Create Your Event :)!</h1>
                   <input type="text" name="state" placeholder="Enter valid state" > 
                         
                   <label for="zip"> Zip </label>
-                  <input type="text" name="zip" placeholder="Enter valid zip"> 
+                  <input type="number" name="zip" placeholder="Enter valid zip">
```
Comment (valvarez1218) [inline src/main/webapp/createEvent.html:51]: Will changing it to a type of number allow for leading zeros?

File: src/main/webapp/createEvent.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+     *                            in the price's value.
+     */
+    date = document.querySelector("form#new-event input[name='date']");
+    price = document.querySelector("form#new-event span input[name='price']")
+    decimalIndex = price.value.indexOf('.');
+
+    date.type = 'text'; //datatime-local type rejects the epoch value.
+    date.value = Date.parse(date.value); //Converts to epoch.
+
+    if(decimalIndex == -1){
```
Comment (valvarez1218) [inline src/main/webapp/createEvent.js:24]: Can you explain what decimalIndex is and why we're checking if it's -1?

**Review by valvarez1218 — COMMENTED** [review]
The code looks good overall, great job! Just had a question about a possible typo/overlook in CreateEventServlet.java and a question in createEvent.js

File: src/main/java/com/google/sps/servlets/CreateEventServlet.java (hunk 1 above)
Comment (josh-hdz) [reply src/main/java/com/google/sps/servlets/CreateEventServlet.java:42]: I changed the request value to match the post request attribute but I kept the variable's name. It does makes it confusing, I'll change it.

File: src/main/webapp/createEvent.html (hunk 2 above)
Comment (josh-hdz) [reply src/main/webapp/createEvent.html:51]: The input's value is always a string (at least in all of our inputs). The attribute "type" only limits the characters allowed and the format. The leading zero will not go away.

File: src/main/webapp/createEvent.js (hunk 3 above)
Comment (josh-hdz) [reply src/main/webapp/createEvent.js:24]: `decimalIndex` is the position at which the first dot character is found in the string (price). If there is no dot character (meaning there is no decimal part) `indexOf()` returns -1. Creating the need to add 2 decimal digits (.00). The other if checks that the price does not have more than 2 decimals if so it rounds it. Java Throws an error when converting a string like "20" to a double data type.

Comment (josh-hdz) [reply src/main/webapp/createEvent.js:24]: Now that I think about it I didn't test when the user inputs a price like "20."; I'll check that tomorrow.

File: src/main/java/com/google/sps/servlets/CreateEventServlet.java (hunk 1 above)
Comment (jmuzy-sps) [reply src/main/java/com/google/sps/servlets/CreateEventServlet.java:42]: You don't need to change if for this, but I do think in general, keeping naming consistent is nice. With this approach, it's easier to see what's coming from the frontend and what's going into the backend, but overall, I think it's just easier to name everything the same

**Review by jmuzy-sps — APPROVED** [review]
Nice job!

**Pull request merged**